src/analysis/ROOM/ROOM.m

```diff
@@ -0,0 +1,130 @@
+function [fluxROOM, solutionROOM, totalFluxDiff] = ROOM(model, fluxWT, rxnKO, varargin)
+% Performs a MILP version of the ROOM (Regulatory on/off minimization of 
+% metabolic flux changes) approach
+%
+% USAGE:
+%
+%    [fluxROOM, solutionROOM, totalFluxDiff] = ROOM(model, WTflux, rxnKO, delta, epsilon, printLevel)
+%
+% INPUTS:
+%    model:            Metabolic model
+%    fluxWT:           Numeric array with flux distribution of wild type
+%    rxnKO:            List of perturbations performed to the model
+%                      (reactions that are eliminated)
+%
+% OPTIONAL INPUTS:
+%    delta:            Multiplicative tol for flux change (Default = 0.03)
+%    epsilon:          Additive tolerance for flux change (Default = 0.001)
+%    printLevel:       Verbose output (Default = 1)
+%
+% OUTPUTS:
+%    fluxROOM:         Flux distribution after ROOM calculation
+%    solutionROOM:     Solution structure
+%    totalFluxDiff:    Euclidean distance of ROOM objective, i.e.
+%                      :math:`\sum (v_{wt}-v_{del})^2`
+% 
+% Solve the following problem:
+%
+% .. math::
+%      min ~&~ \sum y_{i} \\
+%          ~&~ S_{del}v_{del} = 0 \\
+%          ~&~ lb_{del} \leq v_{del} \leq ub_{del} \\
+%          ~&~ for i=1:nRxns\\
+%          ~&~ v_{i} - y_{i}(v_{max,i}-w_{wt,i}^u) \leq w_{wt,i}^u \\
+%          ~&~ v_{i} - y_{i}(v_{min,i}-w_{wt,i}^l) \geq w_{wt,i}^l \\
+%          ~&~ y_{i} \in {0,1} \\
+%          ~&~ w_{wt,i}^u = w_{wt,i} + \delta |w_{wt,i}| + \epsilon \\
+%          ~&~ w_{wt,i}^l = w_{wt,i} - \delta |w_{wt,i}| - \epsilon \\
+%
+% NOTE::
+%
+%    The code here has been based on:
+%    Shlomi, T., Berkman, O., & Ruppin, E. (2005). Regulatory on/off 
+%    minimization of metabolic flux changes after genetic perturbations.
+%    Proceedings of the National Academy of Sciences, 102(21), 7695-7700
+%
+% .. Authors:
+%       - Luis V. Valcarcel, 23/01/2019, University of Navarra, CIMA & TECNUN School of Engineering.
+
+
+p = inputParser;
+% check required arguments
+addRequired(p, 'model');
+addRequired(p, 'WTflux', @(x)isnumeric(x)&&isvector(x));
+addRequired(p, 'rxnKO', @(x)iscell(x));
+% Check optional arguments
+addOptional(p, 'delta', 0.03, @(x)isnumeric(x)&&isscalar(x));
```

Comment:
is it intentional, that these are positioned optional arguments instead of parameters?

Reply:
Thanks @tpfau for the comment. When I first wrote the code I considered they were more used a, as they are part of the mathematical model. I will change them to paramenters.